src/main/java/seedu/address/commons/core/index/Index.java

```diff
@@ -64,6 +89,9 @@ public boolean equals(Object other) {
 
     @Override
     public String toString() {
+        if (this.equals(Index.WILDCARD)) {
```

Comment:
Maybe using isWildcard() is better here since you already have that method in the class?